media_service/admin.py

```diff
@@ -11,26 +11,53 @@ class ResourcesInline(admin.StackedInline):
     verbose_name = 'Resource'
     model = Resource
 
+    def get_queryset(self, request):
+        qs = super(ResourcesInline, self).get_queryset(request)
+        return qs.select_related('media_store', 'course', 'owner')
+
 class CollectionResourcesInline(admin.StackedInline):
     extra = 0
     verbose_name = 'CollectionResource'
     model = CollectionResource
 
 class MediaStoreAdmin(admin.ModelAdmin):
     list_display = ('id', 'file_name', 'file_md5hash', 'file_type', 'file_size', 'img_width', 'img_height', 'reference_count')
+    search_fields = ('file_name', 'file_md5hash')
 
 class CourseAdmin(admin.ModelAdmin):
-    list_display = ('id', 'lti_context_id', 'lti_tool_consumer_instance_guid')
+    list_display = ('id', 'lti_context_id', 'lti_tool_consumer_instance_guid', 'title')
+    search_fields = ('title', 'lti_context_id')
     inlines = (ResourcesInline, CollectionsInline)
 
 class CollectionAdmin(admin.ModelAdmin):
     list_display = ('id', 'title', 'course', 'sort_order')
     ordering = ('course', 'sort_order')
+    search_fields = ('title',)
     inlines = (CollectionResourcesInline,)
 
+    def get_queryset(self, request):
+        qs = super(CollectionAdmin, self).get_queryset(request)
+        return qs.select_related('course')
```

Comment:
Not sure what the interface looks like, but is this something you want to add to `CollectionsInline` as well, like you did with `ResourcesInline`? (And, for that matter, `CollectionResourcesInline`?)